test/e2e/validation_test.go

```diff
@@ -334,3 +337,64 @@ func testExpectedServicesRunning(t *testing.T) {
 		})
 	}
 }
+
+// testCSRApproval tests if the BYOH CSR's have been approved by WMCO
+func testCSRApproval(t *testing.T) {
+	testCtx, err := NewTestContext()
+	require.NoError(t, err)
+	if gc.numberOfBYOHNodes == 0 {
+		t.Skip("BYOH CSR approval testing disabled")
+	}
+	for _, node := range gc.byohNodes {
+		csrs, err := testCtx.findNodeCSRs(node.GetName())
+		require.NotEqual(t, len(csrs), 0, "could not find BYOH node CSR's")
+		require.NoError(t, err, "could not find BYOH node CSR's")
+
+		for _, csr := range csrs {
+			isWMCOApproved := func() bool {
+				for _, c := range csr.Status.Conditions {
+					if c.Type != certificates.CertificateApproved || c.Reason != "WMCOApproved" {
+						return false
+					}
+				}
+				return true
+			}()
+			assert.Equal(t, isWMCOApproved, true, "expected BYOH node CSR to be approved by WMCO: %s", node.GetName())
+		}
+	}
+
+}
+
+// findNodeCSRs returns the list of CSRs for the given node
+func (tc *testContext) findNodeCSRs(nodeName string) ([]certificates.CertificateSigningRequest, error) {
+	var nodeCSRs []certificates.CertificateSigningRequest
+	csrs, err := tc.client.K8s.CertificatesV1().CertificateSigningRequests().List(context.TODO(),
+		meta.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get CSR list")
+	}
+	for _, c := range csrs.Items {
+		parsedCSR, err := parseCSR(&c)
+		if err != nil {
+			return nil, err
+		}
+		dnsAddr := strings.TrimPrefix(parsedCSR.Subject.CommonName, "system:node:")
+		if dnsAddr == "" {
+			return nil, err
+		}
+		if dnsAddr == nodeName {
+			nodeCSRs = append(nodeCSRs, c)
+		}
+	}
+	return nodeCSRs, nil
+}
+
+// parseCSR extracts the CSR from the API object and decodes it.
+func parseCSR(csr *certificates.CertificateSigningRequest) (*x509.CertificateRequest, error) {
+	// extract PEM from request object
+	block, _ := pem.Decode(csr.Spec.Request)
+	if block == nil || block.Type != "CERTIFICATE REQUEST" {
+		return nil, fmt.Errorf("PEM block type must be CERTIFICATE REQUEST")
+	}
+	return x509.ParseCertificateRequest(block.Bytes)
+}
```

Comment:
Since this is repeated code from the CSR package you could export it there + write a unit test and re-use it here